variables.tf

```diff
@@ -83,4 +343,84 @@ locals {
       }
     )
   }
+  kubernetes_cluster = {
+    for kubernetes_cluster in keys(var.kubernetes_cluster) :
+    kubernetes_cluster => merge(
+      local.kubernetes_cluster_values[kubernetes_cluster],
+      {
+        for config in [
+          "api_server_access_profile",
+          "auto_scaler_profile",
+          "azure_active_directory_role_based_access_control",
+          "confidential_computing",
+          "http_proxy_config",
+          "identity",
+          "ingress_application_gateway",
+          "key_management_service",
+          "key_vault_secrets_provider",
+          "kubelet_identity",
+          "linux_profile",
+          "maintenance_window",
+          "microsoft_defender",
+          "monitor_metrics",
+          "oms_agent",
+          "workload_autoscaler_profile",
+          "service_principal",
+          "storage_profile",
+          "web_app_routing"
+        ] :
+        config => merge(local.default.kubernetes_cluster[config], local.kubernetes_cluster_values[kubernetes_cluster][config])
+      },
+      {
+        for config in ["default_node_pool"] :
+        config => merge(
+          merge(local.default.kubernetes_cluster[config], local.kubernetes_cluster_values[kubernetes_cluster][config]),
+          {
+            for subconfig in ["kubelet_config", "node_network_profile", "upgrade_settings"] :
+            subconfig => merge(local.default.kubernetes_cluster[config][subconfig], lookup(local.kubernetes_cluster_values[kubernetes_cluster][config], subconfig, {}))
+          },
+          {
+            for subconfig in ["linux_os_config"] :
+            subconfig => merge(
+              merge(local.default.kubernetes_cluster[config][subconfig], lookup(local.kubernetes_cluster_values[kubernetes_cluster][config], subconfig, {})),
+              {
+                for subsubconfig in ["sysctl_config"] :
+                subsubconfig => merge(local.default.kubernetes_cluster[config][subconfig][subsubconfig], lookup(local.kubernetes_cluster_values[kubernetes_cluster][config][subconfig], subsubconfig, {}))
```

Comment:
@michaelamattes Here I get this error
```
│ Error: Invalid index
│
│   on .terraform/modules/kubernetes_cluster/variables.tf line 388, in locals:
│  388:                 subsubconfig => merge(local.default.kubernetes_cluster[config][subconfig][subsubconfig], lookup(local.kubernetes_cluster_values[kubernetes_cluster][config][subconfig], subsubconfig, {}))
│     ├────────────────
│     │ local.kubernetes_cluster_values is object with 1 attribute "environment"
│
│ The given key does not identify an element in this collection value.
```

The problem comes likely from the fact that there is no linux_os_config block in my local variables. Lookup cannot search for sysctl_config in the linux_os_config block because linux_os_config doesn't exist in local.kubernetes_cluster_values[kubernetes_cluster][config]

Reply:
I've fixed the merge issue. In my tests the examples/min_main.tf and examples/full_main.tf are now working as expected.